packages/fxa-auth-server/scripts/recorded-future/check-and-reset.ts

```diff
@@ -102,122 +123,169 @@ Limit: ${resultsPerPageLimit}
+    return [];
+  }
+
   statsd.increment(
     'recorded-future.identity.credentials-search.results.total',
     leakedLogins.length
   );
 
-  if (leakedLogins.length === 0) {
-    return 0;
+  const { accounts, has2faCount } = await getAccountsByLogin(leakedLogins);
+  if (accounts.size) {
+    statsd.increment(
+      'recorded-future.identity.credentials-search.results.account-emails',
+      accounts.size
+    );
   }
-
-  const log = logFn({ name: 'recorded-future' });
-  const Token = TokenFn(log, config);
-  const Password = PasswordFn(log, config);
-  const DB = createDB(config, log, Token);
-  const authDb = await DB.connect(config, undefined);
-
-  const getAccountRecord = authDb.accountRecord.bind(authDb);
-  const findAccount = createFindAccountFn(getAccountRecord);
-  const getTotpToken = authDb.totpToken.bind(authDb);
-  const hasTotp2FA = createHasTotp2faFn(getTotpToken);
-
-  const accountLogins: SearchResultIdentity[] = [];
-  const accounts: Record<
-    SearchResultIdentity['login'],
-    Awaited<ReturnType<typeof findAccount>>
-  > = {};
-  for (const identity of leakedLogins) {
-    const acct = await findAccount(identity.login);
-
-    if (acct) {
-      const has2FA = await hasTotp2FA(acct);
-
-      if (!has2FA) {
-        accounts[identity.login] = acct;
-        accountLogins.push(identity);
-        statsd.increment(
-          'recorded-future.identity.credentials-search.results.account-emails'
-        );
-      } else {
-        statsd.increment(
-          'recorded-future.identity.credentials-search.results.2fa-accounts'
-        );
-      }
-    }
+  if (has2faCount) {
+    statsd.increment(
+      'recorded-future.identity.credentials-search.results.2fa-accounts',
+      has2faCount
+    );
   }
 
-  if (accountLogins.length === 0) {
-    console.log('No account emails found.');
-    return 0;
+  if (accounts.size === 0) {
+    return [];
   }
 
+  const accountsByLogin: Record<
+    SearchResultIdentity['login'],
+    Awaited<ReturnType<ReturnType<typeof createFindAccountFn>>>
+  > = Array.from(accounts.entries()).reduce((acc, val) => {
+    acc[val[0].login] = val[1];
+    return acc;
+  }, {});
+
   const credentialsLookupFn = createCredentialsLookupFn(client);
-  const cleartextCredentials = await credentialsLookupFn(accountLogins, {
-    first_downloaded_gte: firstDownloadedDateIsoString,
-  });
+  const cleartextCredentials = await credentialsLookupFn(
+    Array.from(accounts.keys()),
+    {
+      first_downloaded_gte: firstDownloadedDate,
+    }
+  );
   statsd.increment(
     'recorded-future.identity.credentials-lookup.cleartext-total',
     cleartextCredentials.length
   );
 
+  const db = await getAuthDb();
   const verifyPassword = createVerifyPasswordFn(
     Password,
-    authDb.checkPassword.bind(authDb)
+    db.checkPassword.bind(db)
   );
 
-  const defaultConcurrency = 100;
-  const concurrency = program.concurrency || defaultConcurrency;
-  const queue = new PQueue({
```

Comment:
I removed this since the password verification and the reset+email steps are broken up so that we can go directly from email args to reset+email.  And upon giving it more thought, I don't think we need it.  Over the course of development, the number of total found logins (that is, before password validation) are fairly low.  And we are going to run this script daily, so I think we can Keep It Simple.